tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -0,0 +1,2789 @@
+        case Stmt::ExprWithCleanupsClass:
+          // Assert that there is a single child of the ExprWithCleanupsClass.
+          assert(++actualExpr->child_begin() == actualExpr->child_end());
+          return Create(*actualExpr->child_begin(), context);
+
+        default:
+          llvm::outs() << "Unknown expression type : " << actualExpr->getStmtClassName() << "\n ";
+          actualExpr->dump(llvm::outs(), context);
+          return nullptr;
+      }
+      // else if (isa<CXXNamedCastExpr>(actualExpr))
+      //{
+      //   return std::make_unique<AstNamedCastExpr>(cast<CXXNamedCastExpr>(actualExpr), context);
+      // }
+      // else if (isa<CastExpr>(actualExpr))
+      //{
+      //   return std::make_unique<AstCastExpr>(cast<CastExpr>(actualExpr), context);
+      // }
+    }
+    else
+    {
+      assert(isa<Stmt>(statement));
+      llvm::outs() << "Unknown statement type : " << statement->getStmtClassName() << "\n ";
+      statement->dump(llvm::outs(), context);
+      return nullptr;
+    }
+  }
+  else
+  {
+    return nullptr;
+  }
+
+  // if (isa<Expr>(statement))
+  //{
+  //   auto expression = cast<Expr>(statement);
+  //   const Expr* actualExpr = expression->IgnoreUnlessSpelledInSource();
+  //   if (isa<IntegerLiteral>(actualExpr))
+  //   {
+  //     return std::make_unique<AstIntExpr>(actualExpr, context);
+  //   }
+  //   else if (isa<StringLiteral>(actualExpr))
+  //   {
+  //     return std::make_unique<AstStringExpr>(cast<StringLiteral>(actualExpr), context);
+  //   }
+  //   else if (isa<FloatingLiteral>(actualExpr))
+  //   {
+  //     return std::make_unique<AstFloatExpr>(cast<FloatingLiteral>(actualExpr), context);
+  //   }
+  //   else if (isa<CXXBoolLiteralExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstBoolExpr>(cast<CXXBoolLiteralExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<CXXNamedCastExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstNamedCastExpr>(cast<CXXNamedCastExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<ImplicitCastExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstImplicitCastExpr>(cast<ImplicitCastExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<CastExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstCastExpr>(cast<CastExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<CXXDefaultInitExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstDefaultInitExpr>(cast<CXXDefaultInitExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<CXXDefaultArgExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstDefaultArgExpr>(cast<CXXDefaultArgExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<CXXConstructExpr>(actualExpr))
+  //   {
+  //     return std::make_unique<AstCtorExpr>(cast<CXXConstructExpr>(actualExpr), context);
+  //   }
+  //   else if (isa<DependentScopeDeclRefExpr>(actualExpr))
```

Comment:
Is this commented code going to come back?  If not we shouldn't checkin commented code

Reply:
It might come back, I'm not 100% sure. But I can reduce its size.